### PR TITLE
[benchmark] Support two modes of consensus

### PR DESCRIPTION
**Pull request by lxfind**
## Description 

Try to exercise the consensus path in the single node benchmark.
There are two modes: noop and direct.

## Test Plan 

CI

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-single-node-benchmark/src/mock_consensus.rs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        consensus_mode: ConsensusMode,
+    ) {
+        let epoch_store = validator.epoch_store_for_testing().clone();
+        let checkpoint_service = Arc::new(CheckpointServiceNoop {});
+        let counter = IntCounter::new(
+            "skipped_consensus_txns",
+            "Total number of consensus transactions skipped",
+        )
+        .unwrap();
+        while let Some(tx) = tx_receiver.recv().await {
```
Comment (mwtian) [inline crates/sui-single-node-benchmark/src/mock_consensus.rs:58]: Not sure what would be the best way to model this, but in practice every consensus commit includes ~1s worth of transactions, so the cost of `process_consensus_transactions` is amortized. Calling the function on every transaction may increase its cost to unrealistic level. Maybe poll every 1s from tx_receiver instead, or poll until a batch size threshold is reached or tx_receiver becomes empty?

File: crates/sui-single-node-benchmark/src/command.rs [hunk 2]
```diff
@@ -68,13 +69,16 @@ pub enum Command {
...
     Baseline,
     /// On top of Baseline, this schedules transactions through the transaction manager.
     WithTxManager,
     /// This goes through the `handle_certificate` entry point on authority_server, which includes
     /// certificate verification, transaction manager, as well as a noop consensus layer.
-    ValidatorService,
+    ValidatorWithNoopConsensus,
+    /// Similar to ValidatorWithNoopConsensus, but the consensus layer would verify the transaction
+    /// and store the sequenced transactions into the store.
+    ValidatorWithDirectConsensus,
```
Comment (mwtian) [inline crates/sui-single-node-benchmark/src/command.rs:83]: I think we can mention this only runs part of consensus handler, or name this `ValidatorWithFakeConsensus`.

**Review by mwtian — APPROVED** [review]

File: crates/sui-single-node-benchmark/src/mock_consensus.rs (hunk 1 above)
Comment (lxfind) [reply crates/sui-single-node-benchmark/src/mock_consensus.rs:58]: Added a parameter to group transactions in a commit. Will add arg flags to update it in future PRs (need some more restructuring of the commands first)

**Comment by lxfind** [comment]
fyi @halfprice once this PR merges you may need to rebase

**Pull request merged**